base/db/R/get.new.site.R

```diff
@@ -0,0 +1,121 @@
+        # No DB connection present. Generate a new ID using one of below steps:
+
+        if (is.null(site$id) | is.na(site$id)) {
+            if ((!is.null(site$lat) && !is.null(site$lon)) &&
+                (!is.na(site$lat) && !is.na(site$lon))
+            ) {
+                site.id <- paste0(site$lat, "_", site$lon)
+                new.site <- data.frame(
+                    id = as.numeric(site.id),
+                    lat = site$lat,
+                    lon = site$lon
+                )
+                str_ns <- paste0(new.site$lat, "_", new.site$lon)
+            } # lat/lon present but ID is absent
+            # Use Pre-existing normalised lat/lon
+            else {
+                PEcAn.logger::logger.warn("Site dataframe does not have an id column")
+                site.id <- generate_site_id()
+                PEcAn.logger::logger.info(paste0("Generated siteID:", site.id))
+                # Optionally, create a new site data frame with the random ID
+                new.site <- data.frame(
+                    id = site.id,
+                    lat = NULL,
+                    lon = NULL
+                )
+                str_ns <- paste0(new.site$id %/% 1e+09, "_", new.site$id %% 1e+09)
+                # We used a different str_ns as identifier here due to absence of lat/lon
+            }
+            # ID as well as lat/lon absent.
+            # Return a WARN as we will be unable to identify such an instance due to lack of information.
+            # We'll try to Generate a new ID similar to previous ones.
+        } else {
+            if ((!is.null(site$lat) && !is.null(site$lon)) &&
+                (!is.na(site$lat) && !is.na(site$lon))
+            ) {
+                new.site <- data.frame(
+                    id = as.numeric(site$id),
+                    lat = site$lat,
+                    lon = site$lon
+                )
+                str_ns <- paste0(new.site$lat, "-", new.site$lon)
+            } # siteId as well as lat/lon present
+            else {
+                new.site <- data.frame(
+                    id = site$id,
+                    lat = NULL,
+                    lon = NULL
+                )
+                str_ns <- paste0(new.site$id %/% 1e+09, "-", new.site$id %% 1e+09)
+            } # siteId present but lat/lon absent
+        }
+    } else {
+        # Check if site dataframe has an id column
+        if (is.null(site$id) | is.na(site$id)) {
+            PEcAn.logger::logger.warn("Site dataframe does not have an id column. Generating a unique ID")
+            if ((!is.null(site$lat) && !is.null(site$lon)) &&
+                (!is.na(site$lat) && !is.na(site$lon))
+            ) {
+                PEcAn.logger::logger.info(paste0("Generated siteID using lat and lon:", site.id))
+                site.id <- generate_site_id(site$lat, site$lon)
+                new.site <- data.frame(
+                    id = as.numeric(site.id),
```

Comment:
does this id need to be numeric? If so, why? Is the algorithm generating it guaranteed to produce numbers?

Reply:
It won't be numeric if using digest as is.